### PR TITLE
Add import timestamp to announcements

### PR DESCRIPTION
**Pull request by andreasphil**
Implements RISDEV-7047. For now I decided to set the timestamp purely on the DB level, i.e. it is set automatically to the current time when the announcement is created. The application can read it, but it can not set the value on insertion or change it afterwards. This should be all we need and minimizes code/API surface.

Coincidentally also implements RISDEV-7048, but I added an integration test to verify that the timestamp is set correctly.

### CODE REVIEW

File: backend/src/main/resources/db/migration/V0.33__add_timestemp_to_announcements_table.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE announcements
```
Comment (VictorDelCampo) [inline backend/src/main/resources/db/migration/V0.33__add_timestemp_to_announcements_table.sql]: type in filename, timestemp --> timestamp

File: backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/output/database/DBServiceIntegrationTest.java [hunk 2]
```diff
@@ -274,7 +274,11 @@ void itFindsAnnouncementOnDB() {
     );
 
     // Then
-    assertThat(announcementOptional).isPresent().contains(announcement);
+    assertThat(announcementOptional)
+      .get()
+      .usingRecursiveComparison()
+      .ignoringFields("importTimestamp")
```
Comment (VictorDelCampo) [inline backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/output/database/DBServiceIntegrationTest.java:282]: Even if we are not setting the timestamp in the application and letting the db create that value I would include a "checking" to some level in an integration test, like maybe just checking that the value is not empty? Or even checking if the value is within the range we would expact (minutes wise?)

File: backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/output/database/DBServiceIntegrationTest.java [hunk 3]
```diff
@@ -302,7 +306,9 @@ void itLoadsAllAnnouncementsFromDB() {
     final List<Announcement> announcements = dbService.loadAllAnnouncements();
 
     // Then
-    assertThat(announcements).containsExactly(announcement2, announcement1);
+    assertThat(announcements)
+      .usingRecursiveFieldByFieldElementComparatorIgnoringFields("importTimestamp")
```
Comment (VictorDelCampo) [inline backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/output/database/DBServiceIntegrationTest.java:315]: Same here as the comment above, though if you check it in the other test, maybe is not needed here

File: backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/input/restapi/AnnouncementControllerIntegrationTest.java [hunk 4]
```diff
@@ -127,6 +130,33 @@ void itCreatesANewAnnouncement() throws Exception {
         .isPresent();
     }
 
+    @Test
+    void itStoresTheImportTimestamp() throws Exception {
```
Comment (VictorDelCampo) [inline backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/input/restapi/AnnouncementControllerIntegrationTest.java:134]: Not sure why a new test (even integration) is needed in the adapter layer without having touched yet the adapter layer... meaning we should the controller is not returning the timestamp yet because the mapper between domain and controller hasn't been touched yet.

**Review by VictorDelCampo — COMMENTED** [review]
Good work, just some comments and possibly the one about the controller test leads to more work on the same branch? I though the ticket would be to develop the whole flow including the API. If not, new ticket needed and i wouldn't include the controller test in this work.

File: backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/input/restapi/AnnouncementControllerIntegrationTest.java (hunk 4 above)
Comment (andreasphil) [reply backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/input/restapi/AnnouncementControllerIntegrationTest.java:134]: I added it because of the way we sliced the ticket. "RISDEV-7048 Extend use case for creating announcements to store the timestamp" is automatically solved by setting the timestamp on the DB level, so nothing to do here. But it's also still expected behavior of the import that the timestamp exists, hence the test.

Actually doing something with the value and returning it in the controller is part of another ticket. Will leave it in for now, especially since otherwise it would be added in the next step anyway.

Comment (VictorDelCampo) [reply backend/src/test/java/de/bund/digitalservice/ris/norms/integration/adapter/input/restapi/AnnouncementControllerIntegrationTest.java:134]: Yeah but the test would be then something else, since for now testing the timestamp you are retrieving the object directly through the repository. When you adapt the mapper from domain to api response schema you will just test the response schema of the API call in the test
